team-manager/src/main/java/ua/knu/knudev/teammanager/service/DepartmentService.java

```diff
@@ -1,21 +1,43 @@
 package ua.knu.knudev.teammanager.service;
 
+import jakarta.validation.Valid;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+import ua.knu.knudev.knudevcommon.utils.AcademicUnitsIds;
 import ua.knu.knudev.teammanager.domain.Department;
+import ua.knu.knudev.teammanager.domain.Specialty;
 import ua.knu.knudev.teammanager.repository.DepartmentRepository;
-import ua.knu.knudev.knudevcommon.utils.AcademicUnitsIds;
+import ua.knu.knudev.teammanagerapi.api.DepartmentApi;
 import ua.knu.knudev.teammanagerapi.exception.DepartmentException;
+import ua.knu.knudev.teammanagerapi.request.DepartmentCreationRequest;
 
 import java.util.Optional;
 import java.util.UUID;
 
 @Service
 @RequiredArgsConstructor
-public class DepartmentService {
+@Validated
+public class DepartmentService implements DepartmentApi {
 
     private final DepartmentRepository departmentRepository;
 
+    @Override
+    public void createDepartment(@Valid DepartmentCreationRequest departmentCreationRequest) {
+        Department department = new Department();
```

Comment:
check here in another method `existsById(UUID id)` that department already exists, in that method throw department exception

Reply:
done